### PR TITLE
DT-1232: 🎨 Switch manage auth accounts to service based and fix no mf…

### PR DESCRIPTION
**Pull request by petergphillips**
…a error

### CODE REVIEW

File: helm_deploy/hmpps-auth/templates/_envs.tpl [hunk 1]
```diff
@@ -149,7 +149,7 @@ env:
     value: {{ include "app.joinListWithComma" .Values.mfa_allowlist | quote }}
 
   - name: APPLICATION_AUTHENTICATION_MFA_ROLES
-    value: "ROLE_MFA,ROLE_PECS_COURT,ROLE_PECS_SUPPLIER,ROLE_PF_STD_PRISON,ROLE_PF_STD_PRISON_RO,ROLE_PF_STD_PROBATION,ROLE_PF_STD_PROBATION_RO,ROLE_PF_POLICE,ROLE_PF_HQ,ROLE_PF_PSYCHOLOGIST,ROLE_PF_LOCAL_READER,ROLE_PF_NATIONAL_READER,ROLE_PPM_ANALYST,ROLE_AUTH_GROUP_MANAGER,ROLE_MAINTAIN_ACCESS_ROLES,ROLE_MAINTAIN_ACCESS_ROLES_ADMIN,ROLE_MAINTAIN_OAUTH_USERS"
+    value: "ROLE_MFA,ROLE_PECS_COURT,ROLE_PECS_SUPPLIER,ROLE_PF_STD_PRISON,ROLE_PF_STD_PRISON_RO,ROLE_PF_STD_PROBATION,ROLE_PF_STD_PROBATION_RO,ROLE_PF_POLICE,ROLE_PF_HQ,ROLE_PF_PSYCHOLOGIST,ROLE_PF_LOCAL_READER,ROLE_PF_NATIONAL_READER,ROLE_PPM_ANALYST,ROLE_AUTH_GROUP_MANAGER"
```
Comment (petergphillips) [inline helm_deploy/hmpps-auth/templates/_envs.tpl:152]: leave group managers turned on for now until PECS turn on service based

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/AbstractMfaController.kt [hunk 2]
```diff
@@ -36,8 +36,8 @@ abstract class AbstractMfaController(
       .addAllObjects(extraModel)
     if (smokeTestEnabled) modelAndView.addObject("smokeCode", mfaData.code)
     modelAndView
-  } catch (e: LockingAuthenticationProvider.MfaUnavailableException) {
-    ModelAndView("redirect:/$startPageUrl", "error", "mfaunavailable")
+  } catch (e: MfaUnavailableException) {
+    ModelAndView("redirect:$startPageUrl", "error", "mfaunavailable")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/AbstractMfaController.kt:40]: fix url

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/MfaClientNetworkService.kt [hunk 3]
```diff
@@ -18,10 +16,6 @@ class MfaClientNetworkService(
 
   private val ipMatchers: List<IpAddressMatcher> = allowlist.map { IpAddressMatcher(it) }
 
-  companion object {
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/MfaClientNetworkService.kt:21]: not used

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/MfaClientService.kt [hunk 4]
```diff
@@ -13,10 +11,6 @@ open class MfaClientService(
   private val mfaClientNetworkService: MfaClientNetworkService,
 ) {
 
-  companion object {
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/MfaClientService.kt:16]: not used

File: src/main/resources/templates/landing.html [hunk 5]
```diff
@@ -6,30 +6,63 @@
...
+         role="alert"
+         data-module="govuk-error-summary">
+      <h2 class="govuk-error-summary__title" id="error-summary">
+        There is a problem
+      </h2>
+      <div class="govuk-error-summary__body" id="error-detail">
+        <ul class="govuk-list govuk-error-summary__list">
+          <li th:each="error: ${param.error}" th:switch="${error}">
+            <a th:case="'mfainvalid'" href="#username"
+               th:text="#{login.mfainvalid(${@environment.getProperty('application.authentication.lockout-count')})}">
```
Comment (petergphillips) [inline src/main/resources/templates/landing.html:21]: not sure need all these, will have to check in a later PR

**Review by mikehalmamoj — APPROVED** [review]

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Pull request merged**